consensus/parlia/snapshot.go

```diff
@@ -426,10 +426,13 @@ func (s *Snapshot) nextProposalBlock(proposer common.Address) (uint64, uint64, e
 	currentProposer := validators[s.Number/uint64(s.TurnLength)%uint64(len(validators))]
 	currentIndex := s.indexOfVal(currentProposer)
 	expectIndex := s.indexOfVal(proposer)
+	if currentIndex < 0 {
```

Comment:
currentIndex := s.Number/uint64(s.TurnLength)%uint64(len(validators))
so `if currentIndex < 0` is impossible